libs/domain/site/src/services/store/default-store.service.ts

```diff
@@ -31,10 +31,14 @@ export class DefaultStoreService implements StoreService {
       switchMap((store) =>
         !store
           ? this.getAll().pipe(
-              map(
-                (stores) =>
-                  stores.find((store) => store.id === this.store) ?? null
-              )
+              map((stores) => {
+                const store = !this.store
+                  ? stores?.[0]
+                  : stores.find((store) => store.id === this.store) ??
+                    stores?.[0];
+                this.store$.next(store);
+                return this.store$.value;
```

Comment:
It would be safer to `return EMPTY` here I believe, as `this.store$.next(store);` will trigger the whole stream anyway.